kong/db/dao/plugins/go.lua

```diff
@@ -131,16 +137,18 @@ do
         end
       end
 
-      if data[1] == 1 and data[2] == my_msg_id then
-        -- it's our answer
-        if data[3] ~= nil then
-          c:setkeepalive()
-          return nil, data[3]
-        end
+      assert(data[1] == 1, "RPC response expected from Go plugin server")
```

Comment:
there can (in theory) be more than one notification before the response.  perhaps something like
```
   if data[1] == 1 then
      assert(data[2] == my_msg_id, "wrong id")
```
would be safer.

Reply:
Wouldn't the extra notifications be caught by the `if data[1] == 2 then` branch already so it will never reach this point?

Reply:
that would get one notification.  if there's a second one (i know, we don't send any), then the assert would fail.

Reply:
that's why it's a loop and not a single condition